### PR TITLE
Fetch full message body when getting email data

### PR DESCRIPTION
**Pull request by 4i60r**

### CODE REVIEW

**Comment by davidsertillange-optimizely** [comment]
That was exactly what I was looking for and fixes issue #294. Thanks a lot!

File: README.md [hunk 1]
```diff
@@ -373,10 +376,20 @@ the data for the specified id is returned instead of the email currently visible
...
+
+Same as above, but returns full HTML email body for each message in thread.
+(Sends two requests instead of one)
+
+
 #### gmail.get.email_data(email_id=undefined, callback)
 
+Does the same as above but accepts a callback function
+
+#### gmail.get.email_data_full(email_id=undefined, callback)
```
Comment (josteink) [inline README.md]: Shouldn't this be get.email_data_full_async ?

**Comment by josteink** [comment]
Sorry about the extremely slow response.

Apart from the comment about the documentation, it seems from the linked issue that this commit may not reliably fix this issue.

As such, having some clear instructions on how to reproduce the error-condition, which this PR could be tested against would be nice.

(I'm saying this, because for me email_data() and email_source() gets me all the stuff I need)

**Comment by andy-sal** [comment]
Unfortunately I don't have any clear instructions but for me, it seemed to be associated with the is_deleted flag. It appeared to truncate the messages to 80 characters when it was set to true (even though emails weren't deleted). From memory it was only in large email threads where past emails were collapsed when the thread was opened. 

I managed to fix the issue we were having by specifying an email ID for the 'msgs' in the query string (as oppose to only 'th') in the api.helper.get.email_data_pre function (see line below).

`url = window.location.origin + window.location.pathname + '?ui=2&ik=' + api.tracker.ik + '&rid=' + api.tracker.rid + '&view=cv&th=' + email_id + '&msgs=' + email_id + '&mb=0&rt=1&search=mbox';`

Hope this helps!

**Comment by josteink** [comment]
Fair enough.

In that case, I think a method rename would be in order `full` makes it sounds like does more than the thing which isn't. This will probably lead many people chosing that one "just in case". Don't you agree?

Would using a identifier more in terms with the error-condition be an option, like `api.get.email_data_including_deleted`?

if you can fix that and also the `async` documentation-issue I observed, I'll be glad to merge this PR.

**Comment by michi88** [comment]
Oh sorry, didn't actually see this. Doesn't this PR actually fix the whole problem? 

https://github.com/KartikTalwar/gmail.js/pull/329

**Comment by josteink** [comment]
@4i60r : With [this PR](https://github.com/KartikTalwar/gmail.js/pull/329) now merged, could you check if this is still an issue?

If not, this PR probably isn't needed and should be closed.

**Comment by 4i60r** [comment]
@josteink It looks like it's fixed. Thanks!

**Comment by josteink** [comment]
Great. Thanks for the quick feedback! I'll go ahead and close this pull request then, since it's no longer needed. 

Anyhow thank you for your contribution. patches are always welcome

**Pull request closed without merging**